app/assets/javascripts/stories.js

```diff
@@ -1,18 +1,23 @@
 document.addEventListener("input", (e) => {
-  if (e.target.classList.contains("story-description")) {
-    const desc = e.target;
-    const form = desc.closest("form");
-    const preview = form.querySelector(".story_preview .content");
-    if (preview) {
-      Rails.ajax({
-        type: "POST",
-        url: "/stories/render_markdown",
-        data: `markdown=${encodeURIComponent(desc.value)}`,
-        dataType: "text",
-        success: (response) => {
-          preview.innerHTML = response;
-        },
-      });
+  function preview_setup(text_area_class, preview_class) {
+    if (e.target.classList.contains(text_area_class)) {
+      const desc = e.target;
+      const form = desc.closest("form");
+      const preview = form.querySelector(preview_class  + " .content");
+      if (preview) {
+        Rails.ajax({
+          type: "POST",
+          url: "/stories/render_markdown",
+          data: `markdown=${encodeURIComponent(desc.value)}`,
+          dataType: "text",
+          success: (response) => {
+            preview.innerHTML = response;
+          },
+        });
+      }
     }
   }
+
+  preview_setup("story-description", ".story_preview")
+  preview_setup("story-extra-info", ".extra_info_preview")
```

Comment:
maybe now thta we use this preview in more than one place we can do it in a more generic way?

I imagine something like using a `data-has-preview` attribute on the textareas and a `data-preview-target` attribute with the selector of the preview element, and then the javascript can be really generic like `document. querySelectorAll('[data-has-preview]').forEach(....)` so we don't have to maintain the classes both in the html and the js? what do you think?

Reply:
@arielj  I need to admit that I'm not much of a frontend dev -- so I took a crack at this, and it works, but I'm not confident there isn't a better way to implement this.  If you have any reccos lemme know

Reply:
I spoke too soon... this seems to work in the browser but it broke the test and I can't figure out why... I might need an assist here

Reply:
we can pair maybe today if you want (though now I know your complicated working hours xD)